### ISSUE TITLE
Kokkos Initialization for Apps using DTK

### ISSUE THREAD

**Issue by sslattery** [ORNL-CEES/DataTransferKit#166]
With Kokkos now being added to DTK Kokkos must be initialized and finalized such as:

```
kokkos::initialize(argc,argv);
// do stuff with DTK
kokkos::finalize();
```

This works great for libraries and applications already using DTK, but does not work for apps that don't. This includes apps that might use other performance portable tools or those that explicitly program using an MPI+X paradigm. 

How should we handle this case? Ideally DTK should be usable without any knowledge of Kokkos/Trilinos. Perhaps we have a DTK initialize and finalize functions for this case that manages Kokkos initalization and finalization.

This is of high priority for many of our potential users.

**Comment by Rombur** [comment]
We either let the user take care of that or we can do sth similar to what is done in [dealii](http://www.dealii.org/developer/doxygen/deal.II/classUtilities_1_1MPI_1_1MPI__InitFinalize.html). We create a class `DTK_InitFinalize`. In the constructor we check if `kokkos` is initialized (it looks like they have a function `is_initialized`) and if `kokkos` is not initialized, then we initialize it. Then, in the destructor we call `kokkos::finalize` _if_ we initialized `kokkos` ourselves. This way, using `DTK_InitFinalize` doesn't hurt if the user already uses `kokkos`.

**Comment by Rombur** [comment]
Fixed by #242
